src/Nethermind/Nethermind.TxPool/TxBroadcaster.cs

```diff
@@ -305,7 +305,13 @@ void NotifyPeers()
                 _txsToSend.Reset();
             }
 
-            NotifyPeers();
+            // Don't broadcast txs while producing a block as it
+            // allocates a lot and we want to minimize GCs at this time
+            // also head about to change which could change the filter results
+            if (!_headInfo.IsProcessingBlock)
```

Comment:
You can put it behind 'IBackgroundTaskScheduler` which prevent having to add `IsProcessingBlock`.

Reply:
Not sure I understand how that would work